frontend-v2/src/app/interest/generator/generator-form.tsx

```diff
@@ -0,0 +1,264 @@
+'use client'
+
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useQuery } from '@tanstack/react-query'
+import { API_PATH, apiClient } from '@/lib/api'
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { Textarea } from '@/components/ui/textarea'
+import { Button } from '@/components/ui/button'
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from '@/components/ui/select'
+import { Checkbox } from '@/components/ui/checkbox'
+
+type FormValues = {
+  name: string
+  title: string
+  description: string
+  chapterId: string
+  activismInterests: boolean
+  referralApplyName: string
+  referralFriends: boolean
+  referralApply: boolean
+  referralOutlet: boolean
+}
+
+export default function GeneratorForm() {
+  const { data: chapterList, isLoading: chapterListIsLoading } = useQuery({
+    queryKey: [API_PATH.CHAPTER_LIST],
+    queryFn: apiClient.getChapterList,
+  })
+
+  const form = useForm<FormValues>({
+    defaultValues: {
+      name: '',
+      title: '',
+      description: '',
+      chapterId: '',
+      activismInterests: false,
+      referralApplyName: '',
+      referralFriends: false,
+      referralApply: false,
+      referralOutlet: false,
+    },
+  })
+
+  const [output, setOutput] = useState('')
+
+  const onSubmit = (data: FormValues) => {
+    const params = new URLSearchParams()
+    params.append('name', data.name)
+    params.append('title', data.title)
+    params.append('description', data.description)
+    params.append('chapterId', data.chapterId)
+    params.append('showInterests', data.activismInterests.toString())
+    params.append(
+      'referralApply',
+      data.referralApplyName === '' ? 'null' : data.referralApplyName,
+    )
+    params.append('showReferralFriends', data.referralFriends.toString())
+    params.append('showReferralApply', data.referralApply.toString())
+    params.append('showReferralOutlet', data.referralOutlet.toString())
+
+    const url = `https://adb.dxe.io/interest?${params.toString()}`
```

Comment:
did you consider using `window.location` here so that it works fully locally too?